src/veh_interact.cpp

```diff
@@ -749,24 +747,27 @@ void veh_interact::do_refill()
     // Now at least one of "fuel tank" is can be refilled.
     // If we have more that one tank we need to create choosing menu
     if (ptanks.size() > 1) {
-        unsigned int pt_choise;
+        unsigned int pt_choice;
         unsigned int entry_num;
         uimenu fuel_choose;
         fuel_choose.text = _("What to refill:");
-        for( entry_num = 0; entry_num < ptanks.size(); entry_num++) {
-            fuel_choose.addentry(entry_num, true, -1, "%s -> %s",
-                                 ammo_name(vehicle_part_types[ptanks[entry_num]->id].fuel_type).c_str(),
-                                 vehicle_part_types[ptanks[entry_num]->id].name.c_str());
+        for( entry_num = 0; entry_num < ptanks.size(); entry_num++ ) {
+            const std::string fuel_type_name = veh->part_info( ptanks[entry_num] ).fuel_type;
+            const std::string type_name = !fuel_type_name.empty() ? 
+                item::find_type( fuel_type_name )->nname(1) : _("Any liquid");
```

Comment:
Should probably refer to the already contained liquid (if any), so the player can't mix liquids.